src/extension/gogoduck/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java

```diff
@@ -0,0 +1,511 @@
+
+        for (File file : directoryListing) {
+            applySubsetSingleFileNcks(file, module);
+        }
+    }
+
+    private static void postProcess(Path tmpDir, GoGoDuckModule module) throws GoGoDuckException {
+        File[] directoryListing = tmpDir.toFile().listFiles();
+        for (File file : directoryListing) {
+            module.postProcess(file);
+        }
+    }
+
+    private static void aggregate(Path tmpDir, Path outputFile) throws GoGoDuckException {
+        aggregateNcrcat(tmpDir, outputFile);
+    }
+
+    private static void aggregateNcrcat(Path tmpDir, Path outputFile) throws GoGoDuckException {
+        List<String> command = new ArrayList<String>();
+        command.add(ncrcatPath);
+        command.add("-D2");
+        command.add("-4");
+        command.add("-h");
+        command.add("-O");
+
+        File[] directoryListing = tmpDir.toFile().listFiles();
+        if (directoryListing.length == 1) {
+            // Special case where we have only 1 file
+            File file = directoryListing[0];
+            try {
+                if(outputFile.toFile().exists() && outputFile.toFile().isFile()) {
+                    logger.info(String.format("Deleting '%s'", outputFile));
+                    Files.delete(outputFile);
+                }
+                logger.info(String.format("Renaming '%s' -> '%s'", file, outputFile));
+                Files.move(file.toPath(), outputFile);
+            }
+            catch (IOException e) {
+                logger.error(e.toString());
+                throw new GoGoDuckException(String.format("Could not rename result file: '%s'", e.getMessage()));
+            }
+        }
+        else {
+            logger.info(String.format("Concatenating %d files into '%s'", directoryListing.length, outputFile));
+            for (File file : directoryListing) {
+                command.add(file.getAbsolutePath());
+            }
+            command.add(outputFile.toFile().getAbsolutePath());
+
+            // Running ncrcat
+            try {
+                execute(command);
+            }
+            catch (Exception e) {
+                throw new GoGoDuckException(String.format("Could not concatenate files into a single file: '%s'", e.getMessage()));
+            }
+        }
+    }
+
+    private static void aggregateJava(Path tmpDir, Path outputFile) throws GoGoDuckException {
+        // TODO implmenet!
+    }
+
+    private static void updateMetadata(GoGoDuckModule module, Path outputFile) {
+        module.updateMetadata(outputFile);
+    }
+
+    private static void cleanTmpDir(Path tmpDir) {
+        logger.debug(String.format("Removing temporary directory '%s'", tmpDir));
+        try {
+            FileUtils.deleteDirectory(tmpDir.toFile());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static GoGoDuckModule getProfileModule(String profile, String geoserver, String subset, UserLog userLog) {
```

Comment:
I think it'd make sense to move this to the `GoGoDuckModule` class itself.  It's typical to call "factory methods" such as this `newInstance`, so it would look like (up above):

```java
GoGoDuckModule module = GoGoDuckModule.newInstance(profile, geoserver, subset, userLog);
```